Recursion/TowerHanoi.cpp

```diff
@@ -0,0 +1,114 @@
+/*
+@Author Davide Spataro www.davidespataro.it
+
+Tower Of Hanoi
+-------------------
+Solves the tower of hanoi problem ( https://en.wikipedia.org/wiki/Tower_of_Hanoi )
+
+using a recursive approach.
+
+  using a recursive approach.
+
+
+
+Time complexity
+---------------
+O(2^n),where n is the number of disks
+
+Space complexity
+----------------
+O(n), where n is the number of disks 
+(not considering the stack space for the required by recursion). 
+*/
+
+
+
+#include <iostream>
+#include<array>
+#include<vector>
+#include <stack>
+#include <unistd.h>
+using namespace std;
+
+
+
+int ndisks;
+int sleeptime = 9000;
+int nmoves = 0;
+
+//prints the current status of the piles
+void drawHanoi(auto& piles) {
+
+  auto printNtimes = [](char c, int n) {
+    for (int i = 0 ; i < n; i++)
+      printf("%c", c);
+  };
+
+  std::array<int,3> sep= {ndisks, ndisks, ndisks};
+
+  for (int  i = ndisks ; i >= 0; --i) {
+    for (int j = 0; j < 3; j++) {
+      printf(" | ");
+      if (piles[j].size() > (unsigned)i) {
+        int n = piles[j][i] + 1;
+        printNtimes('=', n);
+        printNtimes(' ', sep[j] - n + 1);
+      } else
+        printNtimes(' ', sep[j] + 1);
+
+      printf(" | ");
+    }
+    printf("\n");
+  }
+  printf("\n\n");
+
+  usleep(sleeptime);
```

Comment:
If there is no need for usleep() then I think I can get rid of the drawHanoi function too. It would be unusable withouth a sleep cause it will just execute too fast.